File: src/main/java/com/heekng/celloct/repository/JoinRequestRepository.java
package com.heekng.celloct.repository;

import com.heekng.celloct.entity.JoinRequest;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface JoinRequestRepository extends JpaRepository<JoinRequest, Long> {

    List<JoinRequest> findByShopId(Long shopId);

    List<JoinRequest> findByMemberId(Long memberId);

}

File: src/main/kotlin/com/heekng/celloct/repository/JoinRequestRepository.kt
package com.heekng.celloct.repository

import com.heekng.celloct.entity.JoinRequest
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface JoinRequestRepository : JpaRepository<JoinRequest, Long>, JoinRequestRepositoryCustom {

    @EntityGraph(attributePaths = ["member"])
    fun findWithMemberByShopId(shopId: Long): List<JoinRequest>

}